pkg/plugins/golang/v2/scaffolds/api.go

```diff
@@ -105,34 +99,33 @@ func (s *apiScaffolder) scaffold() error {
 			&patches.EnableWebhookPatch{},
 			&patches.EnableCAInjectionPatch{},
 		); err != nil {
-			return fmt.Errorf("error scaffolding APIs: %v", err)
+			return fmt.Errorf("error scaffolding APIs: %w", err)
 		}
 
 		if err := machinery.NewScaffold().Execute(
 			s.newUniverse(),
 			&crd.Kustomization{},
 			&crd.KustomizeConfig{},
 		); err != nil {
-			return fmt.Errorf("error scaffolding kustomization: %v", err)
+			return fmt.Errorf("error scaffolding kustomization: %w", err)
 		}
-
 	}
 
-	if s.doController {
+	if doController {
 		if err := machinery.NewScaffold(s.plugins...).Execute(
 			s.newUniverse(),
-			&controllers.SuiteTest{WireResource: s.doResource},
-			&controllers.Controller{WireResource: s.doResource},
+			&controllers.SuiteTest{WireResource: doAPI},
+			&controllers.Controller{WireResource: doAPI},
 		); err != nil {
-			return fmt.Errorf("error scaffolding controller: %v", err)
+			return fmt.Errorf("error scaffolding controller: %w", err)
 		}
 	}
 
 	if err := machinery.NewScaffold(s.plugins...).Execute(
 		s.newUniverse(),
-		&templates.MainUpdater{WireResource: s.doResource, WireController: s.doController},
+		&templates.MainUpdater{WireResource: doAPI, WireController: doController},
 	); err != nil {
-		return fmt.Errorf("error updating main.go: %v", err)
+		return fmt.Errorf("error updating main.go: %w", err)
```

Comment:
Could we please revert these changes and keep s.doController and s.doAPI as before?

Reply:
I explained in https://github.com/kubernetes-sigs/kubebuilder/pull/1869#discussion_r540064471 why `doAPI` and `doController` need to be stored before updating the resource.